concepts/cpp.scroll

```diff
@@ -227,6 +227,34 @@ hasVirtualFunctions true
      std::cout << "Llamas eat grass!" << std::endl;
    }
  };
+hasTypeClasses true
+ #include <iostream>
+ #include <string>
+
+ using namespace std;
+
+ // NOTE that in contrast to other typeclass-supporting languages,
+ // concept (= typeclass) implementations
+ // a) MUST be defined BEFORE the concept is introduced and
+ // b) do not reference the concept directly, because the association
+ //    is structural, not nominal.
+ string toTypedJson(int x) {
+   return "{\"type\": \"int\", \"value\": " + to_string(x) + "}";
+ }
+
+ template <typename T>
+ concept ToTypedJson = requires (T a) {
```

Comment:
sidenote: I did not know "concept" was a keyword in cpp. interesting.

Reply:
Yeah it's fairly recent (C++20) and I don't think many people use it yet.